### PR TITLE
Add required ports for Overlay to docs

### LINKED ISSUE #17709: DOC Overlay Network Required TCP/UDP Ports + Cluster Options
Need to document the exact  required TCP/UDP ports to be open in order for overlay networks to work:

engine port (e.g TCP 2376), 
vxlan: UDP 4789
serf: TCP + UDP 7946
key-value store ( e.g for consul TCP 8500) 

Additionally, in `Understand Docker container networks`
need to edit `--cluster-advertise=HOST_IP` and make it `--cluster-advertise=HOST_IP:ENGINE_PORT` or `--cluster-advertise=HOST_INTERFACE:ENGINE_PORT`

### PR DESCRIPTION
**Pull request by dave-tucker**
Fixes #17709

Signed-off-by: Dave Tucker dt@docker.com

### CODE REVIEW

File: docs/userguide/networking/dockernetworks.md [hunk 1]
```diff
@@ -404,16 +404,28 @@ provision the hosts are with Docker Machine.
 
 ![Engine on each host](images/engine_on_net.png)
 
+It is important to make sure that the following ports are open between each 
+of your hosts:
+
+| Protocol | Port | Description           |
+|----------|------|-----------------------|
+| tcp      | 2376 | Docker API            |
```
Comment (mavenugo) [inline docs/userguide/networking/dockernetworks.md]: this is not mandatory for overlay networking.

File: docs/userguide/networking/dockernetworks.md [hunk 2]
```diff
@@ -404,16 +404,28 @@ provision the hosts are with Docker Machine.
...
 Docker Engine for use with `overlay` network. There are two options to set:
 
-| Option                           | Description                                               |
-|----------------------------------|-----------------------------------------------------------|
-| `--cluster-store=PROVIDER://URL` | Describes the location of the KV service.               |
-| `--cluster-advertise=HOST_IP`    | Advertises containers created by the HOST on the network. |
+| Option                             | Description                                               |
+|------------------------------------|-----------------------------------------------------------|
+| `--cluster-store=PROVIDER://URL`   | Describes the location of the KV service.               |
+| `--cluster-advertise=HOST_IP:PORT` | Advertises containers created by the HOST on the network. |
```
Comment (mavenugo) [inline docs/userguide/networking/dockernetworks.md]: when you are at it... can you make it <HOST_IP> or <HOST_IFACE> ?

File: docs/userguide/networking/dockernetworks.md [hunk 3]
```diff
@@ -404,16 +404,27 @@ provision the hosts are with Docker Machine.
...
 Docker Engine for use with `overlay` network. There are two options to set:
 
-| Option                           | Description                                               |
-|----------------------------------|-----------------------------------------------------------|
-| `--cluster-store=PROVIDER://URL` | Describes the location of the KV service.               |
-| `--cluster-advertise=HOST_IP`    | Advertises containers created by the HOST on the network. |
+| Option                                        | Description                                               |
+|-----------------------------------------------|-----------------------------------------------------------|
+| `--cluster-store=PROVIDER://URL`              | Describes the location of the KV service.                 |
+| `--cluster-advertise=HOST_IP|HOST_IFACE:PORT` | Advertises containers created by the HOST on the network. |
```
Comment (mavenugo) [inline docs/userguide/networking/dockernetworks.md]: the description is incorrect : `Advertises containers created by the HOST on the network`
It must be `Advertises the host reachability in the cluster`

I understand this comment doesnt belong to this PR.
But I think it is better to fix it while we are at it.

File: docs/userguide/networking/dockernetworks.md [hunk 4]
```diff
@@ -404,16 +404,27 @@ provision the hosts are with Docker Machine.
 
 ![Engine on each host](images/engine_on_net.png)
 
+It is important to make sure that the following ports are open between each 
+of your hosts:
+
+| Protocol | Port | Description           |
+|----------|------|-----------------------|
+| udp      | 4789 | VXLAN                 |
```
Comment (mavenugo) [inline docs/userguide/networking/dockernetworks.md]: would it be better to call it `Overlay data plane (VXLAN)` ?

**Comment by mavenugo** [comment]
Thanks @dave-tucker . LGTM

File: docs/userguide/networking/dockernetworks.md [hunk 5]
```diff
@@ -404,16 +404,27 @@ provision the hosts are with Docker Machine.
 
 ![Engine on each host](images/engine_on_net.png)
 
+It is important to make sure that the following ports are open between each
```
Comment (thaJeztah) [inline docs/userguide/networking/dockernetworks.md]: I'd remove "to make sure"

**Comment by thaJeztah** [comment]
Thanks @dave-tucker, Mary said she'd have a look at this later, so I'll stop for now to not duplicate effort :)

File: docs/userguide/networking/dockernetworks.md [hunk 6]
```diff
@@ -404,16 +404,27 @@ provision the hosts are with Docker Machine.
...
 Docker Engine for use with `overlay` network. There are two options to set:
 
-| Option                           | Description                                               |
-|----------------------------------|-----------------------------------------------------------|
-| `--cluster-store=PROVIDER://URL` | Describes the location of the KV service.               |
-| `--cluster-advertise=HOST_IP`    | Advertises containers created by the HOST on the network. |
+| Option                                        | Description                                               |
+|-----------------------------------------------|-----------------------------------------------------------|
+| `--cluster-store=PROVIDER://URL`              | Describes the location of the KV service.                 |
+| `--cluster-advertise=HOST_IP|HOST_IFACE:PORT` | Advertises the reachability of a HOST to the cluster.     |
```
Comment (moxiegirl) [inline docs/userguide/networking/dockernetworks.md]: Advertises the connection of a cluster host.

File: docs/userguide/networking/dockernetworks.md (hunk 5 above)
Comment (moxiegirl) [inline docs/userguide/networking/dockernetworks.md]: I'd use the imperative. 

You should open the following ports between each of your hosts.

...table..

You key-value store service may require additional open ports.  Check you vendor's documentation and open the appropriate ports.

**Comment by dave-tucker** [comment]
Addressed @moxiegirl's comments...
Still trying to find the right wording for `cluster-advertise` - I think what I have now is both accurate and easy to understand but open to suggestions

**Comment by thaJeztah** [comment]
@dave-tucker I'm good with merging as-is; the important information is in there. If you come with a better suggestion, feel free to open a follow-up PR. Does that work for you?

**Comment by dave-tucker** [comment]
@thaJeztah that works for me

**Comment by thaJeztah** [comment]
LGTM

ping @moxiegirl for the merge (https://github.com/docker/docker/pull/17883#issuecomment-155876224)

**Comment by moxiegirl** [comment]
LGTM

File: docs/userguide/networking/dockernetworks.md [hunk 7]
```diff
@@ -404,16 +404,26 @@ provision the hosts are with Docker Machine.
...
 Docker Engine for use with `overlay` network. There are two options to set:
 
-| Option                           | Description                                               |
-|----------------------------------|-----------------------------------------------------------|
-| `--cluster-store=PROVIDER://URL` | Describes the location of the KV service.               |
-| `--cluster-advertise=HOST_IP`    | Advertises containers created by the HOST on the network. |
+| Option                                        | Description                                                 |
+|-----------------------------------------------|-------------------------------------------------------------|
+| `--cluster-store=PROVIDER://URL`              | Describes the location of the KV service.                   |
+| `--cluster-advertise=HOST_IP|HOST_IFACE:PORT` | The IP address or interface of the HOST used for clustering |
```
Comment (hairyhenderson) [inline docs/userguide/networking/dockernetworks.md]: FYI - the `|` in `--cluster-advertise=HOST_IP|HOST_IFACE:PORT` gets interpreted as a cell break, and messes up rendering in the docs site (https://docs.docker.com/engine/userguide/networking/dockernetworks/)...

**Comment by mavenugo** [comment]
@moxiegirl can you please help with the issue that @hairyhenderson brought up ?

**Comment by thaJeztah** [comment]
Thanks @hairyhenderson, looking at it now, looks to be a bug in the Markdown parser, so I'm trying to find a workaround

**Pull request merged**